app/utils/retry.py

```
@@ -0,0 +1,73 @@
+        async def wrapper(*args, **kwargs) -> Any:
+            attempt = 1
+            current_delay = delay
+            
+            while attempt <= max_attempts:
+                try:
+                    return await func(*args, **kwargs)
+                except exceptions as e:
+                    if attempt == max_attempts:
+                        logger.error(f"Function {func.__name__} failed after {max_attempts} attempts: {str(e)}")
+                        raise
+                    
```

Ruff (0.11.9)

Blank line contains whitespace

Remove whitespace from blank line

(W293)